apps/aecore/src/aec_dev_reward.erl

```diff
@@ -0,0 +1,122 @@
+-module(aec_dev_reward).
+
+-export([ensure_env/0]). % setup hook
+
+-export([enabled/0,
+         activated/1,
+         beneficiaries/0,
+         allocated_shares/0,
+         total_shares/0,
+         split/3]).
+
+-include_lib("aecontract/include/hard_forks.hrl").
+
+
+-define(ENABLED, false).
+-define(TOTAL_SHARES, 1000). % 100 shares == 10% of the reward
+%%% for: "ak_2uN1CEVGs4D5QnFudhp8dSqrwJEMFiTfwvbCK3SJzVYRDiusfd:100" TODO - use actual pubkey for real account
+-define(BENEFICIARIES, [{<<250,151,56,184,99,123,38,230,217,93,156,146,231,90,209,233,15,203,25,102,32,140,178,207,59,171,81,2,249,197,198,61>>, 100}]).
+
+ensure_env() ->
+    Enabled = cfg(<<"protocol_beneficiaries_enabled">>, ?ENABLED),
+    Benefs0 = cfg(<<"protocol_beneficiaries">>, lists:map(fun unparse_beneficiary/1, ?BENEFICIARIES)),
+    case Enabled andalso parse_beneficiaries(Benefs0) of
+        false ->
+            application:set_env(aecore, dev_reward_enabled, false);
+        {ok, Beneficiaries} ->
+            {_, Shares} = lists:unzip(Beneficiaries),
+            AllocShares = lists:sum(Shares),
+            case AllocShares =< ?TOTAL_SHARES of
+                true ->
+                    application:set_env(aecore, dev_reward_enabled, true),
+                    application:set_env(aecore, dev_reward_allocated_shares, AllocShares),
+                    application:set_env(aecore, dev_reward_beneficiaries, Beneficiaries);
+                false ->
+                    lager:error("sum of shares for protocol beneficiaries is too large (> ~p)",
+                                [?TOTAL_SHARES]),
+                    exit({invalid_protocol_beneficiaries, sum_shares_too_large})
+            end;
+        {error, Reason} ->
+            lager:error("invalid protocol beneficiaries: ~p", [Reason]),
+            exit({invalid_protocol_beneficiaries, Reason})
+    end.
+
+cfg(Key, Default) ->
+    aeu_env:user_config([<<"chain">>, Key], Default).
+
+env(Key, Default) ->
+    aeu_env:get_env(aecore, Key, Default).
+
+enabled() ->
+    env(dev_reward_enabled, ?ENABLED).
+beneficiaries() ->
+    env(dev_reward_beneficiaries, ?BENEFICIARIES).
+total_shares() ->
+    ?TOTAL_SHARES.
+allocated_shares() ->
+    case env(dev_reward_allocated_shares, undefined) of
+        undefined ->
+            AllocShares = lists:foldl(fun ({_, X}, A) -> A + X end, 0, beneficiaries()),
+            application:set_env(aecore, dev_reward_allocated_shares, AllocShares),
```

Comment:
May I suggest to avoid lazily setting this environment variable in a getter? It reduces readability and, I expect, makes setup/teardown difficult.

Is deriving this value each time so slow to need lazily setting it?

Reply:
Unresolving as question seems still applicable to me.